File: framework/src/main/java/wulinpeng/com/framework/base/ui/image/imageloader/GlideImageLoaderStrategy.java
package wulinpeng.com.framework.base.ui.image.imageloader;

import android.content.Context;

import com.bumptech.glide.Glide;

/**
 * @author wulinpeng
 * @datetime: 17/4/9 下午8:15
 * @description:
 */
public class GlideImageLoaderStrategy implements IImageLoaderStrategy {
    @Override
    public void load(Context context, ImageLoadEntity imageLoadEntity) {
        Glide.with(context).load(imageLoadEntity.getUrl()).placeholder(imageLoadEntity.getPlaceHolder()).into(imageLoadEntity.getTarget());
    }
}

File: framework/src/main/java/wulinpeng/com/framework/base/ui/image/imageloader/GlideImageLoaderStrategy.kt
package wulinpeng.com.framework.base.ui.image.imageloader

import android.content.Context
import com.bumptech.glide.Glide

/**
 *
 * @author wulinpeng
 * @datetime: 2019/4/6 8:36 PM
 * @description:
 */
class GlideImageLoaderStrategy: IImageLoaderStrategy {
    override fun load(context: Context, imageLoadEntity: ImageLoadEntity) {
        Glide.with(context).load(imageLoadEntity.url).placeholder(imageLoadEntity.placeHolder).into(imageLoadEntity.target)
    }
}